### PR TITLE
Fix for Dragging Problem #907

### PR DESCRIPTION
**Pull request by DerGuteWolf**
Fix for Dragging Problem #907

### CODE REVIEW

**Comment by danzel** [comment]
Please remove the `dist/` changes from this PR

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	"devDependencies": {
 		"git-rev-sync": "^1.8.0",
 		"happen": "^0.3.1",
-		"jake": "~0.5.16",
```
Comment (danzel) [inline package.json]: Why does jake version change?

File: src/MarkerCluster.js [hunk 2]
```diff
@@ -26,14 +26,16 @@ export var MarkerCluster = L.MarkerCluster = L.Marker.extend({
...
-	getAllChildMarkers: function (storageArray) {
+	getAllChildMarkers: function (storageArray, ignoreDraggedMarker) {
 		storageArray = storageArray || [];
 
 		for (var i = this._childClusters.length - 1; i >= 0; i--) {
 			this._childClusters[i].getAllChildMarkers(storageArray);
 		}
 
 		for (var j = this._markers.length - 1; j >= 0; j--) {
+            if (ignoreDraggedMarker && this._markers[j].__dragStart)
```
Comment (danzel) [inline src/MarkerCluster.js]: tabs for indentation

File: src/MarkerClusterGroup.js [hunk 3]
```diff
@@ -708,10 +708,11 @@ export var MarkerClusterGroup = L.MarkerClusterGroup = L.FeatureGroup.extend({
...
 	_childMarkerDragEnd: function (e) {
-		if (e.target.__dragStart) {
-			this._moveChild(e.target, e.target.__dragStart, e.target._latlng);
-		}
-		delete e.target.__dragStart;
+        var dragStart = e.target.__dragStart;
+        delete e.target.__dragStart;
+		if (dragStart) {
+			this._moveChild(e.target, dragStart, e.target._latlng);
+		}
```
Comment (danzel) [inline src/MarkerClusterGroup.js]: Tabs for indentation

File: src/MarkerCluster.js [hunk 4]
```diff
@@ -26,14 +26,16 @@ export var MarkerCluster = L.MarkerCluster = L.Marker.extend({
 	},
 
 	//Recursively retrieve all child markers of this cluster
-	getAllChildMarkers: function (storageArray) {
+	getAllChildMarkers: function (storageArray, ignoreDraggedMarker) {
```
Comment (danzel) [inline src/MarkerCluster.js]: This will need to be included in the documentation.

**Comment by DerGuteWolf** [comment]
ok, will remove dist changes. I only had them first in for the fiddle wo use it.

File: src/MarkerCluster.js (hunk 4 above)
Comment (DerGuteWolf) [reply src/MarkerCluster.js]: Yes, certainly. Should I  write a proposal?

File: package.json (hunk 1 above)
Comment (DerGuteWolf) [reply package.json]: Honestly, I don't know. IMHO jake itself did the change. I will revert this back.

File: src/MarkerCluster.js (hunk 4 above)
Comment (DerGuteWolf) [reply src/MarkerCluster.js]: I have added a sentence to README.md, also indentation is now fixed

**Comment by danzel** [comment]
Thanks, have tweaked the readme a bit.

**Comment by DerGuteWolf** [comment]
Is now everythink ok with the PR?

**Comment by danzel** [comment]
Could we get a spec test, or if not possible then a manual test in `examples/old-bugs` with instructions on how to verify this behaviour?

Should be the last thing I ask for, thanks!

**Comment by DerGuteWolf** [comment]
I tried to add a manual test, but as with all tests in examples/old-bugs I get these errors, which prevent the tests from working (and me from verifying the added test):
MarkerCluster.js:1 Uncaught SyntaxError: Unexpected token export
MarkerClusterGroup.js:5 Uncaught SyntaxError: Unexpected token export

**Comment by danzel** [comment]
I think for the manual tests you probably need to build the dist/ files (but not commit them) and point at those?
There is a good change they are broken because they don't get used often.

**Comment by DerGuteWolf** [comment]
ok, changed now to use dist files. Works for #907. #808 works now with zoom out, but when zooming in again the dragged marker gets lost sometimes. I would propose to use this PR now and open a new bug for this new problem.

**Comment by DerGuteWolf** [comment]
Is now everything ok? Can I do something to get this PR accepted?

**Comment by danzel** [comment]
Merged and 1.4.1 released.
Zooming out still grabs markers (when they would have normally got clustered), but it is an improvement for sure!
Thanks!

**Pull request merged**